mythril/analysis/modules/dependence_on_predictable_vars.py

```diff
@@ -14,174 +14,180 @@ def __init__(self):
         super().__init__(
             name="Dependence of Predictable Variables",
             swc_id="{} {}".format(TIMESTAMP_DEPENDENCE, PREDICTABLE_VARS_DEPENDENCE),
-            hooks=["COINBASE", "GASLIMIT", "TIMESTAMP", "NUMBER", "BLOCKHASH"],
+            hooks=["CALL", "CALLCODE", "DELEGATECALL", "STATICCALL"],
             description=(
                 "Check for CALLs that send >0 Ether as a result of computation "
                 "based on predictable variables such as block.coinbase, "
                 "block.gaslimit, block.timestamp, block.number"
             ),
+            entrypoint="callback",
         )
+        self._issues = []
 
-    def execute(self, statespace):
+    @property
+    def issues(self):
```

Comment:
Can we get some fun type annotations here, `execute`, `solve`, `_analyze_states`, and in the other functions that might need them?

Reply:
Done